server/src/com/thoughtworks/go/server/materials/ConfigMaterialUpdater.java

```diff
@@ -0,0 +1,135 @@
+                                 MaterialChecker materialChecker,
+                                 ConfigMaterialUpdateCompletedTopic configCompletedTopic,
+                                 MaterialUpdateCompletedTopic topic,
+                                 MaterialService materialService,
+                                 SubprocessExecutionContext subprocessExecutionContext)
+    {
+        this.repoConfigDataSource = repoConfigDataSource;
+        this.materialChecker = materialChecker;
+        this.materialRepository = materialRepository;
+        this.configCompleted = configCompletedTopic;
+        this.topic = topic;
+        this.materialService = materialService;
+        this.subprocessExecutionContext = subprocessExecutionContext;
+
+        this.configCompleted.addListener(this);
+    }
+
+    @Override
+    public void onMessage(MaterialUpdateCompletedMessage message) {
+        Material material = message.getMaterial();
+        //MDU is done using the checkout, it has done db update and stored latest changes
+        // but MUS is still waiting for material updated message on MaterialUpdateCompletedTopic
+        if (LOGGER.isDebugEnabled()) {
+            LOGGER.debug(format("[Config Material Update] Config material update completed for material %s. Starting parse process", material));
+        }
+        try {
+            if(message instanceof MaterialUpdateFailedMessage)
+            {
+                MaterialUpdateFailedMessage failure = (MaterialUpdateFailedMessage)message;
+                LOGGER.warn(String.format("[Config Material Update] Cannot update configuration part because material update has failed. Reason: ",
+                        failure.getReason()));
+            }
+            else {
+                File folder = materialRepository.folderFor(material);
+                MaterialRevisions latestModification = materialRepository.findLatestModification(material);
+                Revision revision = latestModification.firstModifiedMaterialRevision().getRevision();
+
+                MaterialRevision lastParseRevision = getMaterialRevisionAtLastParseAttempt(message);
+                if (lastParseRevision == null) {
+                    //never parsed
+                    UpdateConfigurationFromCheckout(folder, revision, material);
+                } else if (latestModification.findRevisionFor(material.config())
+                        .hasChangedSince(lastParseRevision)) {
+                    // revision has changed. the config files might have been updated
+                    UpdateConfigurationFromCheckout(folder, revision, material);
+                } else {
+                    // revision is the same as last time, no need to parse again
+                }
+            }
+        }
+        finally {
+            // always post the original message further
+            // this will remove material from inProgress in MUS
+            topic.post(message);
+        }
+    }
+
```

Comment:
> So far I wasn't thinking much about other scms, but this makes me think, as of now there is no flyweight folder created for Svn, TFS and P4. The flyweight folder really comes into picture for git and hg alone. updateTo method in Material interface was called only on the agent side so far. With this PR, the server would start checking out these other material types as well, ofcourse for the config-repo alone.

I can't find code that explicitly creates the flyweight folder. I did touch the code that returns path of it. Do you suggest that we should create flyweight directory for svn, TFS and P4? IMO when server will call `updateTo` then `svn checkout` command will create the directory.